### PR TITLE
Data aware scheduling docs edits

### PR DESCRIPTION
**Pull request by lzdanski**
---

Copy edits for 2.9 Data Aware Scheduling feature docs.

### CODE REVIEW

**Comment by lzdanski** [comment]
@sunank200 -  Docs copyediting for https://github.com/apache/airflow/pull/37101. I took a look at the PR's you shared with me, https://github.com/apache/airflow/pull/37771 and https://github.com/apache/airflow/pull/38576, and I think copyedits for this content might be covered by https://github.com/apache/airflow/pull/38505.

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst [hunk 1]
```diff
@@ -51,38 +51,38 @@ In addition to scheduling DAGs based upon time, they can also be scheduled based
 What is a "dataset"?
 --------------------
 
-An Airflow dataset is a stand-in for a logical grouping of data. Datasets may be updated by upstream "producer" tasks, and dataset updates contribute to scheduling downstream "consumer" DAGs.
+An Airflow Dataset is a logical grouping of data. Upstream producer tasks can update datasets, and dataset updates contribute to scheduling downstream consumer DAGs.
```
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst]:
```suggestion
An Airflow dataset is a logical grouping of data. Upstream producer tasks can update datasets, and dataset updates contribute to scheduling downstream consumer DAGs.
```

This should be uniform. (Can be all capitalized instead but you don’t in all other cases.)

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst [hunk 2]
```diff
@@ -51,38 +51,38 @@ In addition to scheduling DAGs based upon time, they can also be scheduled based
...
+You must create datasets with a valid URI. Airflow core and providers define various URI schemes that you can use, such as ``file`` (core), ``postgres`` (by the Postgres provider), and ``s3`` (by the Amazon provider). Third-party providers and plugins might also provide their own schemes. These pre-defined schemes have individual semantics that are expected to be followed.
 
 What is valid URI?
 ------------------
 
-Technically, the URI must conform to the valid character set in RFC 3986. If you don't know what this means, that's basically ASCII alphanumeric characters, plus ``%``,  ``-``, ``_``, ``.``, and ``~``. To identify a resource that cannot be represented by URI-safe characters, encode the resource name with `percent-encoding <https://en.wikipedia.org/wiki/Percent-encoding>`_.
+Technically, the URI must conform to the valid character set in RFC 3986, which is basically ASCII alphanumeric characters, plus ``%``,  ``-``, ``_``, ``.``, and ``~``. To identify a resource that cannot be represented by URI-safe characters, encode the resource name with `percent-encoding <https://en.wikipedia.org/wiki/Percent-encoding>`_.
```
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst:71]: We should probably add a link to the Wikipedia entry on URI somewhere too. https://en.wikipedia.org/wiki/Uniform_Resource_Identifier

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst (hunk 1 above)
Comment (lzdanski) [reply docs/apache-airflow/authoring-and-scheduling/datasets.rst]: switched to lowercase - we have it capitalized when it's used in code examples, seemed like a good way to make a quick distinction.

**Comment by lzdanski** [comment]
Wait on review from @sunank200 before merging!

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst [hunk 3]
```diff
@@ -23,7 +23,7 @@ Data-aware scheduling
 Quickstart
 ----------
 
-In addition to scheduling DAGs based upon time, they can also be scheduled based upon a task updating a dataset.
+In addition to scheduling DAGs based on time, you can also schedule DAGs to run based on when a task updates a dataset.
```
Comment (potiuk) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst:26]: Since it's waiting for @sunank200 - comment here to prevent accidental merge :)

Comment (sunank200) [reply docs/apache-airflow/authoring-and-scheduling/datasets.rst:26]: LGTM

**Review by sunank200 — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**